packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -572,29 +616,55 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
             timeLabel = this.props.unitController.numberTranslator(timeForX);
         }
         const chartWidth = this.chartRef.current.chartInstance.width;
+        const chartHeight = this.chartRef.current.chartInstance.height;
         const arraySize = this.state.xyData.labels.length;
         const index = Math.max(Math.round((xPos / chartWidth) * (arraySize - 1)), 0);
         const points: any = [];
         let zeros = 0;
 
         this.state.xyData.datasets.forEach((d: any) => {
-            // In case there are less data points than pixels in the chart,
-            // calculate nearest value.
-            const yValue = d.data[index];
-            const rounded = isNaN(yValue) ? 0 : (Math.round(Number(yValue) * 100) / 100);
-            // If there are less than 10 lines in the chart, show all values, even if they are equal to 0.
-            // If there are more than 10 lines in the chart, summarise the ones that are equal to 0.
-            if (this.state.xyData.datasets.length <= 10 || rounded > 0) {
-                const point: any = {
-                    label: d.label,
-                    color: d.borderColor,
-                    background: d.backgroundColor,
-                    value: rounded.toString(),
-                };
-                points.push(point);
+            let yValue = 0;
+            let xValue = 0;
+            let invalidPoint = false;
+            if (this.isScatterPlot && d.data.length > 0) {
+                // Find the data point that is the closest to the mouse position
+                const closest = this.closestPoint(d.data, chartWidth, chartHeight);
+                if (closest !== undefined) {
+                    yValue = closest.y;
+                    xValue = closest.x;
+                }
+                else {
+                    invalidPoint = true; // Too far from mouse
+                }
             }
             else {
-                zeros += 1;
+                // In case there are less data points than pixels in the chart,
+                // calculate nearest value.
+                yValue = d.data[index];
+            }
+            const rounded: number = isNaN(yValue) ? 0 : (Math.round(Number(yValue) * 100) / 100);
+            let formatted: string = new Intl.NumberFormat().format(rounded);
+            if (this.isScatterPlot && this.props.unitController.numberTranslator) {
+                const time = this.props.unitController.numberTranslator(BigInt(xValue));
+                formatted = '(' + time + ') ' + formatted;
+                timeLabel = 'Process (time stamp) value';
```

Comment:
It is not a given that the series label is a Process, we don't know what it represents, it depends on the analysis. I wonder if we really need a label in this case anyways?

Also, we might want to ask UX experts (now or later) what is the best format to display in the tooltip:
series (time) value
series value (time)
series: value @ time
something else? and should the formatted string be aligned (table) when there are multiple rows?

Reply:
Changed "Process" to "Series".
Yes, we can change the format of the text as soon as the UX experts choose one.